### ISSUE TITLE
Graceful Start of Apollo Gateway

### ISSUE THREAD

**Issue by ankit-joinwal** [apollographql/apollo-server#4499]
Federation needs to be more graceful with the underlying graphql services. If you deploy changes to the federation server and if one of your underlying services is down or there is a mistake, the Apollo Gateway results in a 500 internal error while starting up. This causes the entire data graph of the company to be unavailable.

The error should be isolated to that underlying/remote schema and keep the rest of the gateway should continue to work fine.

**Comment by Imran99** [comment]
Yes I'm having the same issue, we have production monitoring etc to detect production gateway issues as soon as possible however, the gateway should use the schemas it has to partially serve requests. Otherwise the gateway becomes a single point of failure any time an implementing service accidentally deploys an incompatible schema.

**Comment by Imran99** [comment]
One option for working around this is to use Apollo managed federation. I've create a [serverless plugin](https://github.com/Imran99/serverless-plugin-apollo-graphql-federation) to do this. Still, it would be nice to have a graceful failure mode if you don't want to use managed federation.
